File: Cms/src/main/java/com/api/Cms/clients/PreferencesClient.java
package com.api.Cms.clients;

import java.util.List;
import java.util.Map;

import org.springframework.stereotype.Component;
import org.springframework.web.reactive.function.client.WebClient;

import lombok.RequiredArgsConstructor;

@Component
@RequiredArgsConstructor
public class PreferencesClient {

    private final WebClient preferencesWebClient;

    public List<Map<String, Object>> getUserBehaviors(String userId) {
        try {
            return preferencesWebClient.get()
                    .uri("/api/v1/preferences/behavior")
                    .header("X-User-Id", userId)
                    .retrieve()
                    .bodyToFlux(Map.class)
                    .cast(Map.class)
                    .map(m -> (Map<String, Object>) m)
                    .collectList()
                    .block();
        } catch (Exception e) {
            return List.of();
        }
    }

    public List<Map<String, Object>> getUserPreferences(String userId) {
        try {
            return preferencesWebClient.get()
                    .uri("/api/v1/preferences")
                    .header("X-User-Id", userId)
                    .retrieve()
                    .bodyToFlux(Map.class)
                    .cast(Map.class)
                    .map(m -> (Map<String, Object>) m)
                    .collectList()
                    .block();
        } catch (Exception e) {
            return List.of();
        }
    }
}

File: IaUser/service/client/preferences_client.py
import httpx
from config.settings import PREFERENCES_SERVICE_URL

async def get_user_behaviors(user_id: str) -> list:
    try:
        async with httpx.AsyncClient() as client:
            response = await client.get(
                f"{PREFERENCES_SERVICE_URL}/api/v1/preferences/behavior",
                headers={"X-User-Id": user_id}
            )
            if response.status_code == 200:
                return response.json()
    except Exception as e:
        print(f"Error obteniendo comportamientos: {e}")
    return []

async def get_user_preferences(user_id: str) -> list:
    try:
        async with httpx.AsyncClient() as client:
            response = await client.get(
                f"{PREFERENCES_SERVICE_URL}/api/v1/preferences",
                headers={"X-User-Id": user_id}
            )
            if response.status_code == 200:
                return response.json()
    except Exception as e:
        print(f"Error obteniendo preferencias: {e}")
    return []